dot/network/notifications.go

```diff
@@ -362,18 +328,103 @@ func (s *Service) sendData(peer peer.ID, hs Handshake, info *notificationsProtoc
 
 	// we've completed the handshake with the peer, send message directly
 	logger.Tracef("sending message to peer %s using protocol %s: %s", peer, info.protocolID, msg)
-
-	err := s.host.writeToStream(hsData.stream, msg)
-	if err != nil {
+	if err := s.host.writeToStream(stream, msg); err != nil {
 		logger.Debugf("failed to send message to peer %s: %s", peer, err)
+
+		// the stream was closed or reset, close it on our end and delete it from our peer's data
+		if errors.Is(err, io.EOF) || errors.Is(err, mux.ErrReset) {
+			closeOutboundStream(info, peer, stream)
+		}
```

Comment:
nit should we debug log only if the error is not eof or errreset?

Reply:
I think it's ok to log every time, sometimes it's useful to see if we couldn't send it because the peer closed the stream or such